release-notes/swan-lake-beta4-release-note.md

```diff
@@ -509,6 +509,16 @@ public class CustomRetryManager {
 }
 ```
 
+- Attempting to use an out of range float value where the applicable contextually-expected type is `float` will now result in a compile-time error.
+
+- Spec deviations related to identifying the types of numeric literals have been fixed. 
+
+If the numeric literal does not include the float type suffix or the decimal type suffix and it is not a hex floating point literal the type of the numeric literal will be based on following rules.
+
+1. If the literal is a floating point literal, then the possible basic types in order of preference are `[float, decimal]`; otherwise they are `[int, float, decimal]`.
+2. If there is a contextually expected type C and there is a possible basic type N such that N & C is non-empty, use the most preferred such type.
```

Comment:
Can we simplify this. This seems advance for an average users.

Reply:
Can't we write it in this way?
```
If there is a contextual expected type and there is a basic type int, decimal or float in contextual expected type, Use the most perrered type in the order of int, float or decimal.
```

Reply:
We need to mention the intersection bit IMO (https://github.com/ballerina-platform/ballerina-release/pull/1391#discussion_r749879040).

Also, I don't think it covers the finite type scenario.

```suggestion
2. If there is a contextually-expected type `C` and there is an intersection between `C` and the possible numeric basic types identified above, use the most preferred basic type.
```